homeassistant/components/advantage_air/climate.py

```diff
@@ -209,7 +209,7 @@ def target_temperature(self):
     def hvac_mode(self):
         """Return the current HVAC modes."""
         if self._zone["state"] == ADVANTAGE_AIR_STATE_OPEN:
-            return HVAC_MODE_FAN_ONLY
+            return HVAC_MODE_AUTO
```

Comment:
So `HVAC_MODE_AUTO` is defined in Home Assistant as:

`The device is set to a schedule, learned behavior, AI.`

But from your description, it seems that that is not the case?

Reply:
I had the same problem a while ago, because there are no HVAC_MODE_UNKNOWN you need to state the second best HVAC_MODE_AUTO == "I do not know/care".

You can find that in several integrations, but we really should add HVAC_MODE_UNKNOWN.

Reply:
> HVAC_MODE_AUTO == "I do not know/care".

That is not what it is for and should not be used for that either. We should solve problems, not work around them. Ever.

Reply:
@frenck would you think **HVAC_MODE_HEAT_COOL** might be appropriate instead, since the system is adjusting the zone based on its target temp vs current temp, but the actually heating vs cooling vs auto HVAC mode was decided elsewhere?

To explain a little more, in this system the zone climate entity supports a target temp, and being turned **on** or **off**. The actual HVAC mode is set for all zones. I originally chose FAN_ONLY as this most closely resembled the physicality of what you control, the duct is either pushing out the air or it isn't, but this mode does not support temperature control in the Google Assistant (which makes sense logically), so AUTO, or as I suggested above HEAT_COOL represent that the climate entity is dynamically adjusting the temperature, but its not strictly cooling or strictly heating.

Reply:
`HVAC_MODE_HEAT_COOL` will only work as a range of temperature (and will unlock setting a range in the frontend as well).

Reply:
Is it valid to return a HVAC_MODE that the entity doesn't support being set to? I could return the HVAC_MODE being used by the main system, but only allow it to be changed to OFF or FAN_ONLY. Setting to OFF would return OFF regardless of the main system.

Reply:
> Is it valid to return a HVAC_MODE that the entity doesn't support being set to?

That will cause problems, the returned hvac_mode must be in the hvac_modes list provided

Reply:
I would go for `HEAT_COOL`.

HEAT_COOL expects a range, (at which point should it start heat, at which point should it start cooling), but you could set that to the same value.